packages/block-editor/src/components/rich-text/index.native.js

```diff
@@ -389,6 +381,78 @@ export class RichText extends Component {
 		const keyCode = BACKSPACE; // TODO : should we differentiate BACKSPACE and DELETE?
 		const isReverse = keyCode === BACKSPACE;
 
+		this.lastEventCount = event.nativeEvent.eventCount;
+		this.comesFromAztec = true;
+		this.firedAfterTextChanged = event.nativeEvent.firedAfterTextChanged;
+		const value = this.createRecord();
+		const { replacements, text, start, end } = value;
+		let newValue;
+
+		// Always handle full content deletion ourselves.
+		if ( start === 0 && end !== 0 && end >= value.text.length ) {
+			newValue = remove( value, start, end );
+			this.props.onChange( newValue );
+			this.forceSelectionUpdate( 0, 0 );
+			return;
+		}
+
+		if ( this.multilineTag ) {
+			if ( keyCode === BACKSPACE ) {
+				const index = start - 1;
+
+				if ( text[ index ] === LINE_SEPARATOR ) {
+					const collapsed = isCollapsed( value );
+
+					// If the line separator that is about te be removed
+					// contains wrappers, remove the wrappers first.
+					if ( collapsed && replacements[ index ] && replacements[ index ].length ) {
+						const newReplacements = replacements.slice();
+
+						newReplacements[ index ] = replacements[ index ].slice( 0, -1 );
+						newValue = {
+							...value,
+							replacements: newReplacements,
+						};
+					} else {
+						newValue = remove(
+							value,
+							// Only remove the line if the selection is
+							// collapsed, otherwise remove the selection.
+							collapsed ? start - 1 : start,
+							end
+						);
+					}
+				}
+			} else if ( text[ end ] === LINE_SEPARATOR ) {
+				const collapsed = isCollapsed( value );
+
+				// If the line separator that is about te be removed
+				// contains wrappers, remove the wrappers first.
+				if ( collapsed && replacements[ end ] && replacements[ end ].length ) {
+					const newReplacements = replacements.slice();
+
+					newReplacements[ end ] = replacements[ end ].slice( 0, -1 );
+					newValue = {
+						...value,
+						replacements: newReplacements,
+					};
+				} else {
+					newValue = remove(
+						value,
+						start,
+						// Only remove the line if the selection is
+						// collapsed, otherwise remove the selection.
+						collapsed ? end + 1 : end,
+					);
+				}
+			}
+
+			if ( newValue ) {
+				this.onFormatChange( newValue );
+				return;
+			}
+		}
+
 		const empty = this.isEmpty();
```

Comment:
Should we also move this code from here to the end to a function called on `onDeleteKeyDown` like they do in the web?